### PR TITLE
[Feature] Custom html snippet for flash message (#591)

### PR DESCRIPTION
**Pull request by donrestarone**
solves: https://github.com/restarone/violet_rails/issues/560

## [Feature] Custom html snippet for flash message

Dynamic segments are in flash messages! 🎉 Now you can,

1. use ERB / string interpolation in messages for dynamic segments 
2. Optionally use HTML, CSS, JS if you need heavily customized flash notification

## Screenshots
Bootstrap HTML with ERB example: 
Screen Shot 2022-07-26 at 7 17 56 PM

rendered: 
Screen Shot 2022-07-26 at 7 12 43 PM

## Demo

https://user-images.githubusercontent.com/35935196/180994051-e1753ea7-444d-4631-9b2d-b6d11ea6ae3f.mov

### CODE REVIEW

**Comment by donrestarone** [comment]
@Pralish I'm deploying your branch to Heroku for alpha testing, let me know which order you want to merge your work with regards to this PR: https://github.com/restarone/violet_rails/pull/625

**Comment by donrestarone** [comment]
@Pralish custom code works! But when I try to access the `api_resource` explicitly as `api_resource` or `self`-- its erroring: 
Screen Shot 2022-06-04 at 7 05 20 PM

**Comment by donrestarone** [comment]
@Pralish any updates on this?

**Comment by donrestarone** [comment]
@Pralish Thanks for fixing merge conflicts, we can start UAT once you get CI passing

File: app/models/api_resource.rb [hunk 1]
```diff
@@ -2,6 +2,10 @@ class ApiResource < ApplicationRecord
   include JsonbFieldsParsable
 
   after_initialize :inherit_properties_from_parent
+  
+  after_initialize do
```
Comment (restarone-solutions) [inline app/models/api_resource.rb:6]: @Pralish 
Any specific reason for using `after_initialize` ? Because it gets called even for `Model.new`. We could use `after_find` which gets called only after a record has been fetched from the database.

Comment (alis-khadka) [inline app/models/api_resource.rb:6]: @Pralish 
Any specific reason for using `after_initialize`? Because it gets called even for `Model.new`. We could use `after_find` which is invoked only after the record has been fetched from database.

**Comment by donrestarone** [comment]
Once @Pralish addresses @alis-khadka 's comment here: https://github.com/restarone/violet_rails/pull/662#discussion_r919855553 we will need to add in-line documentation as well

**Comment by donrestarone** [comment]
## Heroku UAT  :test_tube: 
Error HTML Snippet should support
1.  interpolating `current_user`, `current_visit`( if any) + always `api_resource` 
2. plain HTML + CSS 
3. inline javascript (just like a snippet)

**Comment by donrestarone** [comment]
@alis-khadka @Pralish please coordinate amongst the two of you to fix merge conflicts on this branch and [address the UAT findings here](https://github.com/restarone/violet_rails/pull/662#issuecomment-1185054052) (HTML and JS cant be used in HTML snippet)

File: app/models/api_resource.rb (hunk 1 above)
Comment (Pralish) [reply app/models/api_resource.rb:6]: @alis-khadka , It's because we need that for new records as well.

**Comment by Pralish** [comment]
@donrestarone, I have addressed the issues. Thanks

File: app/views/comfy/admin/api_actions/_form.html.haml [hunk 2]
```diff
@@ -115,6 +115,9 @@
         = label_tag 'Subject'
         =  text_field_tag "api_namespace[#{type}_attributes][#{index}][email_subject]", resource.email_subject, { class: "form-control form-control-sm" }
 
+      .form-group
```
Comment (donrestarone) [inline app/views/comfy/admin/api_actions/_form.html.haml:118]: @Pralish is this change necessary? At the moment, I don't understand how its related to the feature

**Comment by donrestarone** [comment]
@Pralish I think we need more test coverage on this feature to cover flows like: 

1. ERB in flash message  (should parse dynamic segments)
2. custom HTML in flash message (should show custom HTML)
3. string only in flash message (should show default toast)
